etl/steps/data/meadow/hmd/2022-12-07/life_tables.py

```diff
@@ -0,0 +1,400 @@
+
+Structure would be as follows:
+
+
+    lt_female
+        ├── fltper_1x1
+        │       ├── Albania.txt
+        │       ├── Algeria.txt
+        ...
+        ├── fltper_1x10
+        ...
+    lt_male
+        ├── mltper_1x1
+        │   ├── Albania.txt
+        │   ├── Algeria.txt
+        ...
+    lt_both
+        ├── bltper_1x1
+        │   ├── Albania.txt
+        │   ├── Algeria.txt
+        ...
+
+"""
+import os
+import re
+import tempfile
+from glob import glob
+from io import StringIO
+from pathlib import Path
+from typing import List, Tuple, cast
+
+import pandas as pd
+from owid import catalog
+from owid.catalog import Dataset, Table, TableMeta
+from owid.catalog.utils import underscore_table
+from owid.datautils.io import decompress_file
+from structlog import get_logger
+
+from etl.helpers import Names
+from etl.snapshot import Snapshot
+from etl.steps.data.converters import convert_snapshot_metadata
+
+log = get_logger()
+
+# naming conventions
+N = Names(__file__)
+NAMESPACE = "hmd"
+SHORT_NAME = "hmd"
+VERSION_SNAPSHOT = "2022-12-07"
+DATASET_SNAPSHOT = f"hmd/{VERSION_SNAPSHOT}/{SHORT_NAME}.zip"
+VERSION_MEADOW = "2022-12-07"
+TNAME = "Human Mortality Database"
+
+# Folder structure
+FOLDER_STRUCTURE = [
+    {
+        "name": "lt_both",
+        "files": ["bltper_1x1", "bltper_1x10", "bltper_1x5", "bltper_5x1", "bltper_5x10", "bltper_5x5"],
+        "sex": "both",
+        "regex": (
+            r"([a-zA-Z\-\s,]+), Life tables \(period {table}\), Total\tLast modified: (\d+ [a-zA-Z]{{3}} \d+);  Methods"
+            r" Protocol: v\d+ \(\d+\)\n\n((?s:.)*)"
+        ),
+    },
+    {
+        "name": "lt_female",
+        "files": ["fltper_1x1", "fltper_1x10", "fltper_1x5", "fltper_5x1", "fltper_5x10", "fltper_5x5"],
+        "sex": "female",
+        "regex": (
+            r"([a-zA-Z\-\s,]+), Life tables \(period {table}\), Females\tLast modified: (\d+ [a-zA-Z]{{3}} \d+); "
+            r" Methods"
+            r" Protocol: v\d+ \(\d+\)\n\n((?s:.)*)"
+        ),
+    },
+    {
+        "name": "lt_male",
+        "files": ["mltper_1x1", "mltper_1x10", "mltper_1x5", "mltper_5x1", "mltper_5x10", "mltper_5x5"],
+        "sex": "male",
+        "regex": (
+            r"([a-zA-Z\-\s,]+), Life tables \(period {table}\), Males\tLast modified: (\d+ [a-zA-Z]{{3}} \d+);  Methods"
+            r" Protocol: v\d+ \(\d+\)\n\n((?s:.)*)"
+        ),
+    },
+]
+# Folders expected
+FOLDERS_EXPECTED = [folder["name"] for folder in FOLDER_STRUCTURE]
+
+# Column renaming
+COLUMNS_RENAME = {
+    "Country": "country",
+    "Year": "year",
+    "Age": "age",
+    "mx": "central_death_rate",
+    "qx": "probability_of_death",
+    "ax": "avg_survival_length",
+    "lx": "num_survivors",
+    "dx": "num_deaths",
+    "Lx": "num_person_years_lived",
+    "Tx": "num_person_years_remaining",
+    "ex": "life_expectancy",
+}
+# Column dtypes
+DTYPES = {
+    "central_death_rate": "Float64",
```

Comment:
I would try to avoid Float64, given that there is [an open bug](https://github.com/pandas-dev/pandas/issues/49818) that is absolutely terrifying, which makes methods like isnull, dropna, and so on, useless in certain circumstances. But if you have checked that in this case you don't have any problem, the go ahead.